src/bam/mod.rs

```diff
@@ -507,9 +506,9 @@ impl HeaderView {
     }
 
     /// Retrieve the textual SAM header as a String
-    pub fn text(&self) -> Result<String, FromUtf8Error> {
+    pub fn text(&self) -> Vec<u8> {
```

Comment:
What about just returning the byte slice here? I don't think we need the reallocation, do we?